pkg/test/framework/api/descriptors/descriptors.go

```diff
@@ -41,6 +41,7 @@ var (
 		Requires: []component.Requirement{
 			&ids.Mixer,
 			&ids.Environment,
+			&ids.Galley,
```

Comment:
@nmittler @ozevren @irisdingbj 

This breaks kubernetes environment test run. I'm writing a test only for k8s environment. test requries apps, which indirectly requires pilot, galley. But we don't install galley in kube env.

in registries.go, we only register galley for native.
```
Native.Register(descriptors.Galley, true, galley.NewNativeComponent)
```

And I don't see a galley implementation for kube environment.

How can we fix this?

Reply:
Another worrying fact is that this means we don't run kube env integration2 tests to discover this...
@hklai is that right?

Reply:
I just realized this issue was known already... but what @ozevren proposes seems involve more changes than I need.

Can I do a if/else dependency management based on environment before galley is available in kube env?